seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/checkpoint/operation/TaskCompletedOperation.java

```diff
@@ -54,12 +59,10 @@ protected void readInternal(ObjectDataInput in) throws IOException {
     }
 
     @Override
-    protected PassiveCompletableFuture<?> doRun() throws Exception {
-        CheckpointCoordinator checkpointCoordinator = ((SeaTunnelServer) getService())
+    public void run() {
```

Comment:
Because `PhysicalPlan` and `CheckPointManager` all run on Coordinator Node, So you can get the taskFuture by `JobMaster.PhysicalPlan.SubPlan.PhysicalVertex.taskFuture` and add you callback function .

Reply:
`PhysicalVertex.taskFuture` is a notification of task group granularity, which does not meet the requirements of checkpoint.